homeassistant/components/light/homekit_controller.py

```diff
@@ -0,0 +1,182 @@
+            if ctype == "on":
+                self._chars['on'] = characteristic['iid']
+                self._on = characteristic['value']
+            if ctype == 'brightness':
+                self._chars['brightness'] = characteristic['iid']
+                self._features |= SUPPORT_BRIGHTNESS
+                self._brightness = characteristic['value']
+            elif ctype == 'color-temperature':
+                self._chars['color_temperature'] = characteristic['iid']
+                self._features |= SUPPORT_COLOR_TEMP
+                self._color_temperature = characteristic['value']
+            elif ctype == "hue":
+                self._chars['hue'] = characteristic['iid']
+                self._features |= SUPPORT_RGB_COLOR
+                self._hue = characteristic['value']
+            elif ctype == "saturation":
+                self._chars['saturation'] = characteristic['iid']
+                self._features |= SUPPORT_RGB_COLOR
+                self._saturation = characteristic['value']
+            elif ctype == "value":
+                self._chars['value'] = characteristic['iid']
+                self._features |= SUPPORT_RGB_COLOR
+                self._value = characteristic['value']
+
+    @property
+    def unique_id(self):
+        """Return the ID of this light."""
+        return self._address
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._on
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        if self._features & SUPPORT_BRIGHTNESS:
+            return (self._brightness * 255 / 100)
+        return self._on * 255
+
+    @property
+    def rgb_color(self):
+        """Return the color property."""
+        if self._features & SUPPORT_RGB_COLOR:
+            red, green, blue = colorsys.hsv_to_rgb(self._hue / 100,
+                                                   self._saturation / 100,
+                                                   self._brightness / 100)
+            return (red * 255, green * 255, blue * 255)
+        return (255, 255, 255)
+
+    @property
+    def color_temp(self):
+        """Return the color temperature."""
+        if self._features & SUPPORT_COLOR_TEMP:
+            return self._color_temperature
+        return 3000
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return self._features
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return True
+
+    @property
+    def assumed_state(self):
+        """Return the assumed state."""
+        return True
+
+    def turn_on(self, **kwargs):
+        """Turn the specified light on."""
+        self._state = True
+
+        rgb = kwargs.get(ATTR_RGB_COLOR)
```

Comment:
While this PR was in progress, we've switched the internal color representation to HS, so you should just be able to get `ATTR_HS_COLOR`